include/mun/runtime.h

```diff
@@ -0,0 +1,275 @@
+#ifndef MUN_RUNTIME_CPP_BINDINGS_H_
+#define MUN_RUNTIME_CPP_BINDINGS_H_
+
+#include <algorithm>
+#include <cstdint>
+#include <functional>
+#include <iostream>
+#include <optional>
+#include <string_view>
+
+#include "mun/invoke_result.h"
+#include "mun/runtime_capi.h"
+
+namespace mun {
+    namespace details {
+        template <typename Arg>
+        constexpr bool valid_arg(const MunTypeInfo& arg_type) {
+            auto output_guid = type_guid<Arg>();
+            return std::equal(std::begin(arg_type.guid.b), std::end(arg_type.guid.b), std::begin(output_guid.b), std::end(output_guid.b));
+        }
+
+        template <typename Arg1, typename Arg2>
+        constexpr bool equal_args() {
+            auto arg1_guid = type_guid<Arg1>();
+            auto arg2_guid = type_guid<Arg2>();
+            return std::equal(std::begin(arg1_guid.b), std::end(arg1_guid.b), std::begin(arg2_guid.b), std::end(arg2_guid.b));
+        }
+
+        template <typename Output, typename... Args>
+        constexpr bool valid_fn(const MunFunctionSignature& signature) noexcept {
+            constexpr auto NUM_ARGS = sizeof...(Args);
+
+            if (signature.num_arg_types != NUM_ARGS) {
+                std::cerr << "Invalid number of arguments. Expected: " << std::to_string(NUM_ARGS) << ". Found: " << std::to_string(signature.num_arg_types) << "." << std::endl;
```

Comment:
This line is very long, maybe distribute over multiple lines?

Reply:
Changed.